src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -931,4 +938,62 @@ public Response getHasBeenDeleted(@Context ContainerRequestContext crc, @PathPar
             return ok(dataFileServiceBean.hasBeenDeleted(dataFile));
         }, getRequestUser(crc));
     }
+
+    /**
+     * @param fileIdOrPersistentId Database ID or PID of the data file.
+     * @param dsVersionString The version of the dataset, such as 1.0, :draft,
+     * :latest-published, etc.
+     */
+    @GET
+    @AuthRequired
+    @Path("{id}/versions/{dsVersionString}/citation")
+    public Response getFileCitationByVersion(@Context ContainerRequestContext crc, @PathParam("id") String fileIdOrPersistentId, @PathParam("dsVersionString") String dsVersionString) {
+        try {
+            DataverseRequest req = createDataverseRequest(getRequestUser(crc));
+            final DataFile df = execCommand(new GetDataFileCommand(req, findDataFileOrDie(fileIdOrPersistentId)));
+            Dataset ds = df.getOwner();
+            // Adapted from getDatasetVersionOrDie
+            DatasetVersion dsv = execCommand(handleVersion(dsVersionString, new Datasets.DsVersionHandler<Command<DatasetVersion>>() {
+
+                boolean includeDeaccessioned = true;
+                boolean checkPermsWhenDeaccessioned = true;
+
+                @Override
+                public Command<DatasetVersion> handleLatest() {
+                    return new GetLatestAccessibleDatasetVersionCommand(req, ds);
+                }
+
+                @Override
+                public Command<DatasetVersion> handleDraft() {
+                    return new GetDraftDatasetVersionCommand(req, ds);
+                }
+
+                @Override
+                public Command<DatasetVersion> handleSpecific(long major, long minor) {
+                    return new GetSpecificPublishedDatasetVersionCommand(req, ds, major, minor, includeDeaccessioned, checkPermsWhenDeaccessioned);
+                }
+
+                @Override
+                public Command<DatasetVersion> handleLatestPublished() {
+                    return new GetLatestPublishedDatasetVersionCommand(req, ds);
+                }
+            }));
+
+            if (dsv == null) {
+                return unauthorized("Dataset version cannot be found or unauthorized.");
+            }
+
+            Long getDatasetVersionID = dsv.getId();
+            FileMetadata fm = dataFileServiceBean.findFileMetadataByDatasetVersionIdAndDataFileId(getDatasetVersionID, df.getId());
+            if (fm == null) {
+                return notFound("File could not be found.");
```

Comment:
I suggest using `Bundle.properties` strings as much as we can. I'm sure we can reuse a string for the "File not found" case in many other areas of the API.

Reply:
Sure, no objection. I'll see what I can do.

Reply:
Done: 316524a45e move English to bundle